rpcd/playbooks/roles/rpc_pre_upgrade/tasks/gather_stats.yml

```diff
@@ -73,3 +73,25 @@
     - status
     - instance-volume-mappings
     - running-instances
+
+- name: Obtain a list of neutron routers
+  shell: |
+    . {{ ansible_env.HOME }}/openrc
+    neutron router-list -f value --column id
+  register: neutron_routers
+  delegate_to: "{{ groups['utility_all'][0] }}"
+
+- name: Check agent assigned to each neutron router
+  shell: |
+    . {{ ansible_env.HOME }}/openrc
+    neutron l3-agent-list-hosting-router -f value --column host {{ item }}
+  with_items: "{{ neutron_routers.stdout_lines }}"
+  register: neutron_routers_agents
+  delegate_to: "{{ groups['utility_all'][0] }}"
+
+- name: Write file containing agent for each router
+  copy:
+    content: "{{ item.stdout|replace('_','-') }}" # we replace here since containers will get renamed during upgrade
+    dest: "/tmp/{{ item.item }}.pre-upgrade"
+  with_items: "{{ neutron_routers_agents.results }}"
+  when: item.stdout != ""
```

Comment:
Yep.  I'm not entirely sure when that is reasonable, but I noticed this in my testing when I did this:

```
root@liberty-aio-utility-container-911bf403:~# neutron l3-agent-list-hosting-router lolzlunch

root@liberty-aio-utility-container-911bf403:~# neutron router-create test-router
Created a new router:
+-------------------------+--------------------------------------+
| Field                   | Value                                |
+-------------------------+--------------------------------------+
| admin_state_up          | True                                 |
| availability_zone_hints |                                      |
| availability_zones      |                                      |
| description             |                                      |
| distributed             | False                                |
| external_gateway_info   |                                      |
| ha                      | False                                |
| id                      | a9619e01-f8b9-4a89-956d-ecb6b9135a6b |
| name                    | test-router                          |
| routes                  |                                      |
| status                  | ACTIVE                               |
| tenant_id               | 50fc51fde749453e810f49d5bc571a82     |
+-------------------------+--------------------------------------+
root@liberty-aio-utility-container-911bf403:~#
```

It looks like until I attach something to that router it remains unscheduled on an agent.